File: solutions/julia/cars-assemble/1/cars-assemble.jl
function success_rate(speed)
    if speed == 0
        0.0
    elseif speed < 5
        1.0
    elseif speed < 9
        0.9
    elseif speed < 10
        0.8
    else
        0.77
    end
end

function production_rate_per_hour(speed)
    221 * speed * success_rate(speed)
end

function working_items_per_minute(speed)
    floor(Int, production_rate_per_hour(speed) / 60)
end

File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {

    private static final int CARS_PER_HOUR = 221;

    private double successRate(int speed) {
        if (speed < 5) {
            return 1;
        } else if(speed < 9) {
            return 0.9;
        } else if(speed < 10) {
            return 0.8;
        } else {
            return 0.77;
        }
    }
    
    public double productionRatePerHour(int speed) {
        return successRate(speed) * speed * CARS_PER_HOUR;
    }

    public int workingItemsPerMinute(int speed) {
        return (int) (productionRatePerHour(speed) / 60); 
    }
}
